### PR TITLE
Add Action For Deploying Android Applications To Google Play Store

### PR DESCRIPTION
**Pull request by pkramaric**
### Checklist

  - [x] PR title is clear and describes the work
  - [x] Commit messages are self-explanatory and summarize the change

### Action

  - [x] Tests are provided/updated for the new/existing action
  - [x] The action maintainer in `CODEOWNERS` is updated
  - [x] The `README.md` file for new/existing action is created/updated

### CODE REVIEW

File: deploy-android-app/action.yml [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+        echo Version Name: ${{ env.RELEASE_VERSION }}
+
+        sed -i "s/versionCode [0-9]*/versionCode $current_version_code/g" app/build.gradle
+        sed -i "s/versionName \"[^\"]*\"/versionName \"${{ env.RELEASE_VERSION }}\"/g" app/build.gradle
+        current_version_code=$((current_version_code+1))
+        echo "$current_version_code" > VERSION_CODE
+        echo "${{ env.NEXT_VERSION }}" > VERSION
+   
+      #JDK 1.8
+    - name: Set up JDK 1.8
```
Comment (srdjanflybits) [inline deploy-android-app/action.yml:116]: is this supposed to be?
```suggestion
    - name: Set up JDK 18
```

File: deploy-android-app/action.yml [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+inputs:
+  github_token:
+    description: A GitHub personal access token with repo scope to access private repositories.
+    required: true
+  branch:
+    description: Which branch is running the job
+    required: true
+    type: choice
+    options:
+      - main
```
Comment (srdjanflybits) [inline deploy-android-app/action.yml:13]: is this used?

File: deploy-android-app/action.yml [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+  steps:
+
+    #Checkout Code
+    - name: Checkout Code
+      uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+        token: ${{ inputs.github_token }}
+
+    #Validate Inputs
```
Comment (srdjanflybits) [inline deploy-android-app/action.yml:55]: move validation to before checkout since checkout uses github_token

Comment (srdjanflybits) [reply deploy-android-app/action.yml:55]: validate all input
```
        [[ "${{ inputs.version }}" ]] || { echo "version input is empty" ; exit 1; }
        [[ "${{ inputs.who_trigger }}" ]] || { echo "who_trigger input is empty" ; exit 1; }
        [[ "${{ inputs.keystore_file }}" ]] || { echo "keystore_file input is empty" ; exit 1; }
        [[ "${{ inputs.alias }}" ]] || { echo "alias input is empty" ; exit 1; }
        [[ "${{ inputs.keystore_password }}" ]] || { echo "keystore_password input is empty" ; exit 1; }
        [[ "${{ inputs.alias_password }}" ]] || { echo "alias_password input is empty" ; exit 1; }
        [[ "${{ inputs.service_account_json }}" ]] || { echo "service_account_json input is empty" ; exit 1; }
        [[ "${{ inputs.package_name }}" ]] || { echo "package_name input is empty" ; exit 1; }
```

File: deploy-android-app/action.yml [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+      #UPLOAD TO PLAY STORE
+    - name: Deploy to Play Store (BETA)
+      id: deploy
+      uses: r0adkll/upload-google-play@v1
+      with:
+        serviceAccountJson: service_account.json
+        packageName: ${{ inputs.package_name }}
+        releaseFiles: ${{steps.sign_app.outputs.signedReleaseFile}}
+        track: internal
+
```
Comment (srdjanflybits) [inline deploy-android-app/action.yml:176]:
```suggestion

```

File: deploy-android-app/action.yml [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+        echo Version Code: $current_version_code
+        echo Version Name: ${{ env.RELEASE_VERSION }}
+
+        sed -i "s/versionCode [0-9]*/versionCode $current_version_code/g" app/build.gradle
+        sed -i "s/versionName \"[^\"]*\"/versionName \"${{ env.RELEASE_VERSION }}\"/g" app/build.gradle
+        current_version_code=$((current_version_code+1))
+        echo "$current_version_code" > VERSION_CODE
+        echo "${{ env.NEXT_VERSION }}" > VERSION
+   
+      #JDK 1.8
```
Comment (srdjanflybits) [inline deploy-android-app/action.yml:115]:
```suggestion

      #JDK 18
```

**Review by srdjanflybits — APPROVED** [review]

File: deploy-android-app/action.yml (hunk 4 above)
Comment (srdjanflybits) [reply deploy-android-app/action.yml:176]: this is just extra tab

File: deploy-android-app/action.yml (hunk 2 above)
Comment (pkramaric) [reply deploy-android-app/action.yml:13]: yes i am suppose to validate that it is the main branch. So I will add that to validation

**Comment by pkramaric** [comment]
Thanks @srdjanflybits I updated the code

**Review by srdjanflybits — APPROVED** [review]
not important but the comment still needs to be updated to #JDK 18

**Pull request merged**